src/plugins/files/server/blob_storage_service/adapters/es/es.ts

```diff
@@ -106,14 +114,32 @@ export class ElasticsearchBlobStorageClient implements BlobStorageClient {
             maxChunkSize: this.chunkSize,
           },
         });
-        await pipeline.apply(null, [src, ...(transforms ?? []), dest] as unknown as Parameters<
-          typeof pipeline
-        >);
 
-        return {
-          id: dest.getContentReferenceId()!,
-          size: dest.getBytesWritten(),
+        const start = performance.now();
+        await pipeline([src, ...(transforms ?? []), dest]);
+        const end = performance.now();
+
+        const _id = dest.getContentReferenceId()!;
+        const size = dest.getBytesWritten();
+
+        const perfArgs: PerformanceMetricEvent = {
+          eventName: FILE_UPLOAD_PERFORMANCE_EVENT_NAME,
+          duration: end - start,
+          key1: 'size',
+          value1: size,
+          meta: {
+            datasource: 'es',
+            id: _id,
+            index: this.index,
+            chunkSize: this.chunkSize,
+          },
         };
+
+        if (analytics) {
+          reportPerformanceMetricEvent(analytics, perfArgs);
+        }
```

Comment:
This is a nit, but would it be possible to not even do the measurement if `analytics` is not present?

--- 

I had one random thought, feel free to push back on this!

Perhaps we could register a singleton value on the plugin class, like

```ts
class FilesPlugin {
  private static analytics: ...
  private static setAnalyitics(a: ...): void {}
  public static getAnalyitics(): undefined | ... {}
}
```

Similar to what we are doing with the `ElasticsearchBlobStorageClient` and the semaphor singleton.

Then your `withReportPerformanceMetric` helper can import and get the `analytics` dependency from there. This would avoid the need to pass the dependency all the way down.

Reply:
This way we could use your wrapper everywhere too :)

Reply:
Sure I have updated it 👍🏽